File: easy/compressed_sequence.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp($line);
    my ( $b, $c ) = ( -1, 1 );
    foreach my $i ( split( / /, $line ) ) {
        if ( $i == $b ) {
            $c++;
        }
        else {
            if ( $b >= 0 ) {
                print "$c $b ";
                $c = 1;
            }
            $b = $i;
        }
    }
    print "$c $b\n";
}

File: easy/compressed_sequence.rb
File.open(ARGV[0]).each_line do |line|
  s = line.split.map(&:to_i)
  b = -1
  c = 1
  s.each do |i|
    if i == b
      c += 1
    else
      if b >= 0
        print c.to_s + ' ' + b.to_s + ' '
        c = 1
      end
      b = i
    end
  end
  puts c.to_s + ' ' + b.to_s
end
